File: ncarg2d/src/libncarg/support/semess.f
C
C $Id: semess.f,v 1.6 2008-07-27 00:17:31 haley Exp $
C                                                                      
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      FUNCTION SEMESS (ITRIM)
C
        CHARACTER*113 SEMESS
C
C The value of this function is some portion of the current error
C message (blanks if there is no current error).  If the value of
C ITRIM is less than or equal to zero, the entire error message is
C returned.  If the value of ITRIM is equal to 1, prepended routine
C names and associated slashes are removed.  If the value of ITRIM
C is 2 or greater, all routine names are removed, along with the
C following hyphen and the blanks around the hyphen.  For example,
C if the current error message is
C
C   'CPCLAM/AREDAM - AREA-MAP ARRAY OVERFLOW'
C
C then
C
C   SEMESS(0)='CPCLAM/AREDAM - AREA-MAP ARRAY OVERFLOW'
C   SEMESS(1)='AREDAM - AREA-MAP ARRAY OVERFLOW'
C   SEMESS(2)='AREA-MAP ARRAY OVERFLOW'
C
C The common blocks SECOMI and SECOMC are used to hold shared variables
C of types INTEGER and CHARACTER, respectively, for the routine SETER
C and associated routines.  For descriptions of these variables and for
C default values of them, see the block data routine SEBLDAX.
C
        COMMON /SECOMI/ IERRU,IERRF,IRECF,LOMSG
        SAVE   /SECOMI/
C
        COMMON /SECOMC/ ERMSG
          CHARACTER*256 ERMSG
        SAVE   /SECOMC/
C
C Return either blanks or the requested portion of the current message.
C
        IF (IERRF.EQ.0) THEN
          SEMESS=' '
        ELSE IF (ITRIM.LE.0) THEN
          SEMESS=ERMSG(1:LOMSG)
        ELSE
          IBEG=1
          IF (ITRIM.EQ.1) THEN
            DO 101 I=1,LOMSG
              IF (ERMSG(I:I).EQ.'/') IBEG=I+1
              IF (ERMSG(I:I).EQ.'-') GO TO 103
  101       CONTINUE
          ELSE
            DO 102 I=1,LOMSG
              IF (ERMSG(I:I).EQ.'-') THEN
                IBEG=I+2
                GO TO 103
              END IF
  102       CONTINUE
          END IF
  103     IF (IBEG.GT.LOMSG) IBEG=1
          SEMESS=ERMSG(IBEG:LOMSG)
        END IF
C
C Done.
C
        RETURN
C
      END

File: ncarg2d/man/error_handling/semess.m
.TH SEMESS 3NCARG "March 1994" UNIX "NCAR GRAPHICS"
.na
.nh
.SH NAME
SEMESS - Called by a user to get a specified portion of the current error
message.
.SH SYNOPSIS
MESSG = SEMESS(ITRIM)
.SH C-BINDING SYNOPSIS
#include <ncarg/ncargC.h>
.sp
char *c_semess(int itrim)
.SH DESCRIPTION 
In any routine that references
SEMESS, it must be declared to be of type CHARACTER*113.
.sp
If the internal error flag of SETER is non-zero, the value of the character
expression "SEMESS(ITRIM)" is the remembered error message associated with
that error flag, trimmed as indicated by the value of the input quantity
ITRIM.  Otherwise, it is a string of blanks.
.sp
When ITRIM = 0 or less, no trimming is done; the whole error message is
returned.  When ITRIM = 1, prepended routine names and the associated slashes
are trimmed off.  When ITRIM = 2 or greater, all leading routine names are
trimmed off; just the error description is returned.
.sp
Example: Assume that the
current error message is the string "CPCLAM/AREDAM - AREA-MAP ARRAY OVERFLOW";
then, SEMESS(0) is that entire string, while SEMESS(1) = "AREDAM - AREA-MAP
ARRAY OVERFLOW", and SEMESS(2) = "AREA-MAP ARRAY OVERFLOW".
.sp
The argument of SEMESS is as follows:
.sp
.IP "ITRIM" 12
(an input expression of type INTEGER) - Says what part of the error message
is to be returned, as detailed above.
.SH C-BINDING DESCRIPTION 
The C-binding argument descriptions are the same as the FORTRAN 
argument descriptions.
.SH EXAMPLES
Use the ncargex command to see the following relevant
examples: 
tseter,
arex02.
.SH ACCESS
To use SEMESS or c_semess, load the NCAR Graphics libraries ncarg, ncarg_gks,
and ncarg_c, preferably in that order. 
.SH SEE ALSO
Online:
entsr, eprin, errof, error_handling, fdum, icfell, icloem, nerro, retsr,
seter, ncarg_cbind
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br
The use of this Software is governed by a License Agreement.
